### PR TITLE
Expand description for AKs in registration_commands controller

### PR DESCRIPTION
**Pull request by Lennonka**
#### What are the changes introduced in this pull request?

Expanded description (doc text) for the `activation_keys` parameter.

#### Considerations taken when implementing this change?

* Requested by RH support for improved clarity of the API endpoint/Hammer command.
* Cherry-pick as far as possible, please

#### What are the testing steps for this pull request?

Only changes inside a string value. Review from TWs and ack from Devs.

### CODE REVIEW

File: app/controllers/katello/concerns/api/v2/registration_commands_controller_extensions.rb [hunk 1]
```diff
@@ -28,7 +28,7 @@ module ApiPieExtensions
         update_api(:create) do
           param :registration_command, Hash do
             param :activation_key, String, desc: N_('Activation key for subscription-manager client, required for CentOS and Red Hat Enterprise Linux. For multiple keys use `activation_keys` param instead.'), deprecated: true
-            param :activation_keys, Array, desc: N_('Activation keys for subscription-manager client, required for CentOS and Red Hat Enterprise Linux. Required only if host group has no activation keys.')
+            param :activation_keys, Array, desc: N_('Activation keys for subscription-manager client, required for CentOS and Red Hat Enterprise Linux. Required only if host group has no activation keys or if host group is not assigned.')
```
Comment (Lennonka) [inline app/controllers/katello/concerns/api/v2/registration_commands_controller_extensions.rb:31]: "or if no host group is given"

Would this be better?

Comment (jeremylenz) [reply app/controllers/katello/concerns/api/v2/registration_commands_controller_extensions.rb:31]: I'm fine with either :+1:

**Review by jeremylenz — COMMENTED** [review]
Thanks @Lennonka!

Will wait a bit before acking to see if this gets any more comments.

**Comment by Lennonka** [comment]
@jeremylenz I've invited the TWs the doc channel to review. We usually wait 24 hours for comments, so I'm okay with that.

**Comment by jeremylenz** [comment]
@Lennonka This needs a Redmine issue; can you please create one at projects.theforeman.org ? (Project should be "Katello")

Once that's done, you'll have to edit the commit message as per [this](https://github.com/Katello/katello/pull/11031/checks?check_run_id=26142595861)

File: app/controllers/katello/concerns/api/v2/registration_commands_controller_extensions.rb [hunk 2]
```diff
@@ -28,7 +28,7 @@ module ApiPieExtensions
         update_api(:create) do
           param :registration_command, Hash do
             param :activation_key, String, desc: N_('Activation key for subscription-manager client, required for CentOS and Red Hat Enterprise Linux. For multiple keys use `activation_keys` param instead.'), deprecated: true
-            param :activation_keys, Array, desc: N_('Activation keys for subscription-manager client, required for CentOS and Red Hat Enterprise Linux. Required only if host group has no activation keys.')
+            param :activation_keys, Array, desc: N_('Activation keys for subscription-manager client, required for CentOS and Red Hat Enterprise Linux. Required only if host group has no activation keys or if you do not provide a host group.')
```
Comment (asteflova) [inline app/controllers/katello/concerns/api/v2/registration_commands_controller_extensions.rb:31]: I think "required for CentOS and RHEL" and "required only if host group has no activation keys" could be somewhat confusing, if not contradictory, statements. Can you think of an alternative word to replace the second instance of "required"? For example:

```suggestion
            param :activation_keys, Array, desc: N_('Activation keys for subscription-manager client, required for CentOS and Red Hat Enterprise Linux. Use only if host group has no activation keys or if you do not provide a host group.')
```

Comment (jeremylenz) [reply app/controllers/katello/concerns/api/v2/registration_commands_controller_extensions.rb:31]: There are two instances where `activation_keys` is not required:

1. The host is not CentOS or RHEL; or
2. The host group already provides an activation key.

So maybe 'required' is fine for both? I'm okay with 'use' as well

**Comment by Lennonka** [comment]
Doc-wise peer reviewed and good to go.

**Review by jeremylenz — APPROVED** [review]
Thanks @Lennonka !

ACK :+1:

**Pull request merged**